upgrade_test.py

```diff
@@ -1111,16 +1114,38 @@ def test_kubernetes_operator_upgrade(self):
         self.assertEqual(expected_docker_image_tag, actual_docker_image_tag)
 
         InfoEvent(message='Step6 - Wait for the update of Scylla cluster').publish()
-        # NOTE: rollout starts with some delay which may take even 20 seconds.
-        #       Also rollout itself takes more than 10 minutes for 3 Scylla members.
-        #       So, sleep for some time to avoid race with presence of existing rollout process.
-        time.sleep(60)
-        self.k8s_cluster.kubectl(
-            f"rollout status statefulset/{self.params.get('k8s_scylla_cluster_name')}-"
-            f"{self.k8s_cluster.region_name}-{self.k8s_cluster.rack_name}"
-            " --watch=true --timeout=20m",
-            timeout=1205,
-            namespace=self.db_cluster.namespace)
+        for scylla_pod in self.db_cluster.nodes[::-1]:
+            scylla_pod.wait_till_k8s_pod_get_uid(
+                ignore_uid=old_scylla_pods_uids[scylla_pod.name], throw_exc=True, timeout=900)
+        for scylla_pod in self.db_cluster.nodes[::-1]:
+            scylla_pod.wait_for_pod_readiness()
+
+        scylla_cluster_conditions = json.loads(self.k8s_cluster.kubectl(
+            f"get scyllacluster {self.params.get('k8s_scylla_cluster_name')} -o json",
+            namespace=self.db_cluster.namespace).stdout.strip())["status"]["conditions"]
+        for condition in scylla_cluster_conditions:
+            assert condition["reason"] != "Error", "'ScyllaCluster' failed to be updated: %s" % condition
```

Comment:
you should check `Available=True,Progressing=False,Degraded=False`
https://github.com/scylladb/scylla-operator/blob/eb144ac50175c040e486f47eb9468908181ce678/pkg/api/scylla/v1/types_cluster.go#L543

Reply:
nit: I assume this could also check that Progressing condition has a last update timestamp > last check before upgrade, to be sure it changed

Reply:
Done